sklearn/impute/tests/test_impute.py

```diff
@@ -1669,6 +1669,12 @@ def test_imputer_transform_preserves_numeric_dtype(dtype_test):
 
 @pytest.mark.parametrize("array_type", ["array", "sparse"])
 @pytest.mark.parametrize("keep_empty_features", [True, False])
+# TODO(1.8): Remove
+@pytest.mark.filterwarnings(
+    "ignore:When `keep_empty_features` is false and `constant` is set the "
+    "column filled with missing values will be removed in 1.8. "
+    "to keep the same behaviour, set `keep_empty_features` to True"
+)
```

Comment:
So we should catch the warning here.

Instead, we should make sure that in the code, a warning is raised when we expect it. It means that I would change slightly the code in the body of the test function with something like:


```python
    for method in ["fit_transform", "transform"]:
        if method.startswith("fit") and not keep_empty_features:
            warn_msg = "column filled with missing values will be removed"
            with pytest.warns(FutureWarning, match=warn_msg):
                X_imputed = getattr(imputer, method)(X)
        else:
            X_imputed = getattr(imputer, method)(X)
        assert X_imputed.shape == X.shape
```

So for the case we are calling `fit_transform` (and thus `fit`) and that `keep_empty_features == False`, then we catch the warning and check that we have the expected warning message. In all other case, we don't make any check. If by error a warning is raised then one of the CI build will fail because it will be transformed into an error because this is an undesired warning.